server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -540,52 +569,26 @@ toAnnGValue =
 
 -- | Extract from the Hasura results the remote relationship arguments.
 extractRemoteRelArguments ::
-     RemoteSchemaMap
-  -> EncJSON
+     EncJSON
   -> [RemoteRelField]
   -> Either GQRespValue ( GQRespValue , [ BatchInputs ])
-extractRemoteRelArguments remoteSchemaMap hasuraJson rels =
-  -- TODO refactor with MonadError
-  case OJ.eitherDecode (encJToLBS hasuraJson) >>= parseGQRespValue of
+-- ^ TODO I think we can just return (GQRespValue, [BatchInputs]) but need to look closer at the callsite.
+extractRemoteRelArguments hasuraJson rels =
+  case parseGQRespValue hasuraJson of
     Left (fromString-> err) ->
       Left $
       appendJoinError emptyResp $ "decode error: " <> err
-    Right gqResp@(GQRespValue mdata _merrors) ->
-      case mdata of
-        Nothing ->
-          Left $
-          appendJoinError gqResp "no hasura data to extract for join"
-        Just value -> do
-          let hashE =
-                execStateT
-                  (extractFromResult One keyedRemotes (OJ.Object value))
-                  mempty
-          case hashE of
-            Left err -> Left $ appendJoinError gqResp err
-            Right hash -> do
-              remotes <-
-                flip Map.traverseWithKey hash $
-                  \key rows ->
-                    case Map.lookup key keyedMap of
-                      Nothing ->
-                        Left $
-                        appendJoinError gqResp "failed to associate remote key with remote"
-                      Just remoteRel ->
-                        case Map.lookup
-                               (rtrRemoteSchema
-                                  (rmfRemoteRelationship
-                                     (rrRemoteField remoteRel)))
-                               remoteSchemaMap of
-                          Just remoteSchemaCtx ->
-                            pure (remoteRel, (rscInfo remoteSchemaCtx), rows)
-                          Nothing ->
-                            Left $
-                              appendJoinError gqResp "could not find remote schema info"
-
-              pure (gqResp, map (\(_, _, batchInputs ) -> batchInputs) $ Map.elems remotes)
+    Right gqResp@GQRespValue{..} -> do
```

Comment:
Hmm, I now see what you mean. This was written by Chris and is another one of those functions which seemed too complicated to delve into at that time.